From df0082fd206c22300e26ff6dda465b166d9b6d9e Mon Sep 17 00:00:00 2001
From: Alexey Prokhin <alexey@prokhin.ru>
Date: Wed, 17 Oct 2018 19:55:27 +0300
Subject: [PATCH 0307/1562] winex11.drv: Ignore clip_reset when trying to clip
 the mouse after the desktop has been resized.

This fixes the mouse clipping when the desktop is resized multiple times in a row.

CW-Bug-Id: #21879
---
 dlls/win32u/input.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/win32u/input.c b/dlls/win32u/input.c
index ff148b4cbd7..28cd0a11677 100644
--- a/dlls/win32u/input.c
+++ b/dlls/win32u/input.c
@@ -2533,7 +2533,7 @@ BOOL clip_fullscreen_window( HWND hwnd, BOOL reset )
     if (!NtUserGetWindowRect( hwnd, &rect )) return FALSE;
     if (!NtUserIsWindowRectFullScreen( &rect )) return FALSE;
     if (is_captured_by_system()) return FALSE;
-    if (NtGetTickCount() - thread_info->clipping_reset < 1000) return FALSE;
+    if (!reset && NtGetTickCount() - thread_info->clipping_reset < 1000) return FALSE;
     if (!reset && clipping_cursor && thread_info->clipping_cursor) return FALSE;  /* already clipping */
 
     if (!(monitor = NtUserMonitorFromWindow( hwnd, MONITOR_DEFAULTTONEAREST ))) return FALSE;
-- 
2.45.2

